cluster/gce/ubuntu/node.yaml

```diff
@@ -0,0 +1,281 @@
+From nobody Tue Aug 11 10:13:54 2015
+Content-Type: multipart/mixed; boundary="===============6024533374511606659=="
+MIME-Version: 1.0
+
+--===============6024533374511606659==
+MIME-Version: 1.0
+Content-Type: text/upstart-job; charset="us-ascii"
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="kube-addons.conf"
+
+#upstart-job
+
+description "Install kubelet add-on manifest files"
+
+start on stopped kube-docker
+
+script
+	# Configuration files are located at /etc/saltbase.
+	. /etc/kube-env
+	if [ "${ENABLE_NODE_LOGGING}" = "true" ]; then
+		if [ "${LOGGING_DESTINATION}" = "gcp" ]; then
+			cp /etc/saltbase/kubernetes/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml /etc/kubernetes/manifests/
```

Comment:
Correct, no salt installed. The purpose of pulling salt tarball is for supporting addons. The addon yaml files are in cluster/saltbase/salt. One way is to upload needed yaml files to GCE metadata server and then downloaded by node instances. But this way is hard to extend. If we have new kubelet addons, we have to change both the shell and node.yaml. Another way is like here, just simply upload the entire saltbase tarball, and nodes can use any yaml file. When we add new kubelet addons, we can simply change this particular upstart job to place the yaml to the right directory.